sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessPipelineRunner.java

```diff
@@ -166,4 +185,159 @@ private InProcessPipelineRunner(InProcessPipelineOptions options) {
   public InProcessPipelineOptions getPipelineOptions() {
     return options;
   }
+
+  @Override
+  public <OutputT extends POutput, InputT extends PInput> OutputT apply(
+      PTransform<InputT, OutputT> transform, InputT input) {
+    Class<?> overrideClass = defaultTransformOverrides.get(transform.getClass());
+    if (overrideClass != null) {
+      transform.validate(input);
+      // It is the responsibility of whoever constructs overrides to ensure this is type safe.
```

Comment:
I'll document the defaultTransformOverrides field; It's actually just about the constructor.